apps/pass-image-api/src/telemetry_conf.rs

```diff
@@ -40,28 +29,34 @@ fn get_resource() -> Resource {
 fn init_tracer() {
     global::set_text_map_propagator(TraceContextPropagator::new());
 
-    let tracer_provider = opentelemetry_otlp::new_pipeline()
-        .tracing()
-        .with_trace_config(
-            opentelemetry_sdk::trace::Config::default().with_resource(get_resource()),
-        )
-        .with_exporter(opentelemetry_otlp::new_exporter().tonic())
-        .install_batch(runtime::Tokio)
-        .expect("Failed to initialise tracing provider");
+    let exporter = opentelemetry_otlp::SpanExporter::builder()
+        .with_tonic()
+        .build()
+        .expect("Failed to create span exporter");
+
+    let tracer_provider = opentelemetry_sdk::trace::SdkTracerProvider::builder()
+        .with_batch_exporter(exporter)
+        .with_resource(get_resource())
+        .build();
 
     global::set_tracer_provider(tracer_provider);
 }
 
 // A Meter Provider is a factory for Meters
 // A Meter creates metric instruments, capturing measurements about a service at runtime.
 fn init_meter_provider() -> Result<()> {
-    let meter_provider = opentelemetry_otlp::new_pipeline()
-        .metrics(opentelemetry_sdk::runtime::Tokio)
-        .with_exporter(opentelemetry_otlp::new_exporter().tonic())
-        .with_resource(get_resource())
-        .with_delta_temporality()
+    let exporter = opentelemetry_otlp::MetricExporter::builder()
+        .with_tonic()
         .build()
-        .with_context(|| "creating meter provider")?;
+        .with_context(|| "creating metric exporter")?;
```

Comment:
Could you explain what `with_context` is doing here?
Just asking because I've never seen it before.

Reply:
Ah, the beauty of overloaded words :D 

This is anyhow's [with_context](https://docs.rs/anyhow/latest/anyhow/trait.Context.html#example) - it lets you add extra info to errors that bubbles up when this one is unwrapped by the `?`. So you get something back that gives you a bit more of a clue of what was happening when it went wrong.